newsparser.py

```diff
@@ -1,62 +1,25 @@
-import email
 import os
-import hashlib
 import datetime
 import traceback
-import html
-from email.policy import EmailPolicy
-from email.headerregistry import HeaderRegistry, UnstructuredHeader
-
-
-class telegramAttachment:
-    def __init__(self, file_name, file_content, file_type):
-        ext = os.path.splitext(file_name)[1].replace('"', '')
-        self.name = hashlib.sha1(file_name.encode('utf-8')).hexdigest() + ext
-        self.content = file_content
-        self.type = file_type
-        self.file_id = None
-
-
-def isPlusOne(msg):
-    return msg.startswith("+1") and len(msg) < 10
-
-
-def tryParseAttachment(part):
-    content_disp = part.get("Content-Disposition", None)
-    if content_disp is None:
-        return None
-    disps = content_disp.strip().split(';')
-    if disps[0].lower() != "attachment":
-        return None
-    file_data = part.get_payload(decode=True)
-    content_type = part.get_content_type()
-    file_name = "whoknows"
-    for config in disps[1:]:
-        param, value = (x.strip() for x in config.split('='))
-        if param == "filename":
-            file_name = value
-            break
-    return telegramAttachment(file_name, file_data, content_type)
-
+import re
 
 class newsArticle:
-    headers = ("From", "Newsgroups", "Subject", "Date")
-    registry = HeaderRegistry()
-    registry.map_to_type('From', UnstructuredHeader)
-    policy = EmailPolicy(header_factory=registry)
-
-    def __init__(self, raw_msg, mention_manager, topic):
+    def __init__(self, user, topic, subject, date, raw_msg, raw_html, mention_manager):
+        self.user = user
+        self.topic = topic
+        self.subject = subject
+        self.date = date
         self.raw_msg = raw_msg
+        self.raw_html = raw_html
         self.mention_manager = mention_manager
-        self.topic = topic
         self.broken = False
         self.content = None
         self.is_plus_one = None
         self.attachments = None
 
     def isPlusOne(self):
         if self.is_plus_one is None:
-            self.parseMessage()
+            self.is_plus_one = self.raw_msg.startswith("+1") and len(self.raw_msg) < 10
```

Comment:
could you rather keep the `isPlusOne` and call it here ?

in case someone tries to improve it in the future, wouldn't need to touch any other parts.